### DISCUSSION TITLE
How can I use UUID with this package?

### DISCUSSION THREAD

**Discussion by DJZT** [osiset/laravel-shopify#641] [Q&A]
@osiset 
In `Osiset\ShopifyApp\Contracts\ShopModel` interface `getShopId()` method expecting the `Osiset\ShopifyApp\Objects\Values\ShopId` object as return type. But this object can be only integer type inside.
And UUID can not be used as ID value.

I think the best solution - use inside `Osiset\ShopifyApp\Contracts\ShopModel` interface `ShopId` interface.

For example `Osiset\ShopifyApp\Contracts\ShopId`. This will give possible to create own implementation of `ShopId`

**Comment by gnikyt** [comment]
Interesting, oversight by me, everything in the ShopModel trait should reference an interface. I can update this.

**Reply by DJZT** [reply]
Cool. It will be great. Will checking the update

**Reply by gnikyt** [reply]
The PR is in and merged. getID method now maps to an interface, so you should be fine overriding it now with your own object when I release an update.

**Reply by DJZT** [reply]
I tried to use the `dev-master` version and found additional issues.
Seems like the next classes required to change to use `Osiset\ShopifyApp\Contracts\Objects\Values\ShopId` instead of `Osiset\ShopifyApp\Objects\Values\ShopId` as minimal requirements to perform the login action

`Osiset\ShopifyApp\Actions\AfterAuthorize`
`Osiset\ShopifyApp\Storage\Commands\Shop`
`Osiset\ShopifyApp\Storage\Queries\Shop`
`Osiset\ShopifyApp\Actions\DispatchScripts`
`Osiset\ShopifyApp\Messaging\Jobs\ScripttagInstaller`
`Osiset\ShopifyApp\Actions\CreateScripts`
`Osiset\ShopifyApp\Actions\DispatchWebhooks`
`Osiset\ShopifyApp\Messaging\Jobs\WebhookInstaller`
`Osiset\ShopifyApp\Actions\CreateWebhooks`
`Osiset\ShopifyApp\Contracts\Commands\Shop`

**Reply by gnikyt** [reply]
Good catch! Master updated, see if this resolves your issue, if so, I can push a patch.

**Reply by DJZT** [reply]
Seems like it is working now. Thanks a lot
